common/src/main/java/io/netty/util/internal/PlatformDependent.java

```diff
@@ -1198,6 +1219,16 @@ private static boolean equalsSafe(byte[] bytes1, int startPos1, byte[] bytes2, i
         return true;
     }
 
+    private static boolean equalsSafeConsistentTime(byte[] bytes1, int startPos1,
+                                                    byte[] bytes2, int startPos2, int length) {
+        byte b = 0;
```

Comment:
I ran benchmarks comparing some equals functions, and found that it is significantly faster to use an int rather than a byte:

```
HeadersBenchmark.arrays              4096  sample  129777  1227.096 ± 4.670  ns/op
HeadersBenchmark.digest              4096  sample  127528  2488.555 ± 6.749  ns/op
HeadersBenchmark.theBytes            4096  sample   99710  3170.983 ± 9.587  ns/op
HeadersBenchmark.theChars            4096  sample  105714  2992.226 ± 8.468  ns/op
HeadersBenchmark.theInts             4096  sample  126811  2497.360 ± 7.106  ns/op
HeadersBenchmark.theLongs            4096  sample   99672  3181.179 ± 8.380  ns/op
HeadersBenchmark.theShorts           4096  sample  100008  3158.683 ± 8.164  ns/op
```

`arrays` is Arrays.equals (for comparison)
`digest` is MessageDigest.isEqual  (broken upto in partly java6, fixed in 2009)
All the others are copy paste versions of MessageDigest.isEqual  using different accumulators.